src/app/project/project-documents/detail/detail.component.ts

```diff
@@ -115,12 +136,28 @@ export class DocumentDetailComponent implements OnInit, OnDestroy {
    * @return {void}
    */
   onEdit() {
-    this.storageService.state.selectedDocs = [this.document];
-    this.storageService.state.labels = this.document.labels;
+    this.storageService.state.selectedDocs = [this.mapRowData(this.document)];
+    this.storageService.state.labels = this.document.labels || [];
     this.storageService.state.back = { url: ['/p', this.document.project, 'project-files', 'detail', this.document._id], label: 'View File' };
-    this.router.navigate(['p', this.document.project, 'project-files', 'edit']);
+		this.router.navigate(['p', this.document.project, 'project-files', (this.document.externalLink ? 'edit-link' : 'edit')]);
   }
 
+	onDownload() {
+		if (this.document.externalLink) {
+			window.open(this.document.externalLink, "_blank");
```

Comment:
WCAG guidelines state that even external links must open in the same tab whenever possible. One of the exceptions is if you're in the middle of filling out a form. Is that the case here? Otherwise, it should open in the same tab.

https://www.w3.org/WAI/WCAG21/Techniques/general/G200

Reply:
I removed some instances of "_blank". I believe a document open happens in a new tab by default though.